### PR TITLE
Makes all Mobs Hard Delete

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Mobs just don't GC well under the vast vast majority of circumstances; they have so many references, it's ridiculous. I see them crop up in the "failed to GC" log way too much; we're just going to hard delete every single one.

### CODE REVIEW

File: code/modules/mob/mob.dm [hunk 1]
```diff
@@ -19,7 +19,8 @@
 		for(var/datum/alternate_appearance/AA in viewing_alternate_appearances)
 			AA.viewers -= src
 		viewing_alternate_appearances = null
-	return ..()
+	.. ()
```
Comment (tigercat2000) [inline code/modules/mob/mob.dm]: 👀 
dat space

**Comment by tigercat2000** [comment]
Which mob type exactly doesn't GC? New players, ghosts? I think a better solution might just be to figure out why they aren't GCing, they are hardly living mobs with their excessive reference count

File: code/modules/mob/mob.dm [hunk 2]
```diff
@@ -19,7 +19,8 @@
...
+	return QDEL_HINT_HARDDEL
```
Comment (tigercat2000) [inline code/modules/mob/mob.dm]: QDEL_HINT_HARDDEL_NOW

File: code/modules/mob/mob.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/modules/mob/mob.dm]: Reee

**Comment by Fox-McCloud** [comment]
They still have a significant amount of references and again, still consistently fail to GC; if you want to fiddle around with the find references thing and manual searching--but I'd rather not have references dangle for these mobs until that's fixed.

**Comment by tigercat2000** [comment]
@Fox-McCloud Again, please specify _which_ non-living mob types fail to GC 😛 
I don't mind this PR being merged, but I think it is worth looking into, if not opening an issue report

**Comment by Fox-McCloud** [comment]
`/mob/new_player` and `/mob/dead/observer`

**Pull request merged**